frontend/src/lib/components/AnnotationsOverlay/AnnotationsOverlay.tsx

```diff
@@ -0,0 +1,248 @@
+            onMouseLeave={() => {
+                setHovered(false)
+                console.log(activeDate?.toISOString(), date.toISOString(), activeDate === date)
+                if (!isDateLocked) {
+                    deactivateDate()
+                }
+            }}
+            onClick={
+                !isDateLocked
+                    ? lockDate
+                    : active
+                    ? unlockDate
+                    : () => {
+                          const button = buttonRef.current as HTMLButtonElement
+                          activateDate(date, [button.offsetLeft, button.offsetTop])
+                      }
+            }
+        >
+            <LemonBubble
+                count={annotations.length || <IconPlusMini className="w-full h-full" />}
+                size="small"
+                style={active && isDateLocked ? { outline: '0.125rem solid var(--primary)' } : undefined}
+            />
+        </button>
+    )
+})
+
+function AnnotationsPopover(): JSX.Element {
+    const { popoverAnnotations, activeDate, intervalUnit, isDateLocked, insightId } = useValues(annotationsOverlayLogic)
+    const { closePopover } = useActions(annotationsOverlayLogic)
+    const { openModalToCreateAnnotation } = useActions(annotationModalLogic)
+
+    return (
+        <LemonModal
+            className="AnnotationsPopover"
+            inline
+            title={`${pluralize(popoverAnnotations.length, 'annotation')} • ${activeDate?.format(
+                INTERVAL_UNIT_TO_HUMAN_DAYJS_FORMAT[intervalUnit]
+            )}`}
+            footer={
+                <LemonButton
+                    type="primary"
+                    onClick={() => openModalToCreateAnnotation(activeDate, insightId)}
+                    disabled={!isDateLocked}
+                >
+                    Add annotation
+                </LemonButton>
+            }
+            closable={isDateLocked}
+            onClose={closePopover}
+        >
+            {popoverAnnotations.length > 0 ? (
+                <div
+                    className="flex flex-col gap-1 w-full overflow-y-auto"
+                    style={{
+                        maxHeight: '28rem',
+                    }}
+                >
+                    {popoverAnnotations.map((annotation) => (
+                        <AnnotationCard key={annotation.id} annotation={annotation} />
+                    ))}
+                </div>
+            ) : (
+                'There are no annotations in this period.'
+            )}
+        </LemonModal>
+    )
+}
+
+function AnnotationCard({ annotation }: { annotation: ParsedAnnotationType }): JSX.Element {
+    const { insightId } = useValues(annotationsOverlayLogic)
+    const { deleteAnnotation } = useActions(annotationsModel)
+    const { openModalToEditAnnotation } = useActions(annotationModalLogic)
+
+    return (
+        <div className="AnnotationCard flex flex-col gap-2 w-full p-3 rounded border border-border">
```

Comment:
border-border is the same as `border` (the default border color is already applied)